ext/date/php_date.c

```diff
@@ -4901,9 +4899,9 @@ static void php_do_date_sunrise_sunset(INTERNAL_FUNCTION_PARAMETERS, int calc_su
 		case 1:
 			retformat = SUNFUNCS_RET_STRING;
 		case 2:
-			latitude = INI_FLT("date.default_latitude");
+			latitude = DATE_DEFAULT_LATITUDE;
 		case 3:
-			longitude = INI_FLT("date.default_longitude");
+			longitude = DATE_DEFAULT_LONGITUDE;
```

Comment:
Having these as ini's still made some small sense in that you could configure your server for your location (if such a thing exists) -- but a hardcoded default is completely useless. If we drop the ini settings, these parameters should be required.

Reply:
I agree that these parameters should not have a default, but they come after the optional $format parameter. :(

Reply:
That would have to become required as well ... people would have had to specify it for any realistic use of the function anyway, to get at the lat/long parameters.

Reply:
Okay, but in this case we should have deprecated relying on the defaults in PHP 7.4. Guess it's too late now. Postpone? @derickr ?

Reply:
A bit unfortunate that this didn't make it into the 7.4 deprecations. I guess this isn't particularly critical though and we can deprecate for 8.0?

Reply:
Given that this topic has been brought up 12 years ago, I don't think it's critical, so deprecating for 8.0 would be fine for me.

Reply:
To make sure we don't forget about it, I've created https://wiki.php.net/rfc/deprecations_php_8_0 and added a stub for this. If anyone wants to fill out the argument, feel free :)